### PR TITLE
[core] Upgrade monorepo

### PR DESCRIPTION
**Pull request by cherniavskii**
To support live demo editing introduced in https://github.com/mui/material-ui/pull/34454

Preview: https://deploy-preview-6549--material-ui-x.netlify.app/x/react-data-grid/layout/

- [ ] backported to `master` branch

### CODE REVIEW

File: docs/next.config.js [hunk 1]
```diff
@@ -43,6 +43,14 @@ module.exports = withDocsInfra({
     return {
       ...config,
       plugins,
+      resolve: {
```
Comment (cherniavskii) [inline docs/next.config.js:46]: See https://github.com/mui/material-ui/pull/32107#issuecomment-1180889892
I don't really understand why this change is necessary

File: .eslintrc.js [hunk 2]
```diff
@@ -77,6 +77,7 @@ module.exports = {
     'jsdoc/require-returns': ['error', { contexts: ['TSFunctionType'] }],
     'jsdoc/require-returns-type': ['error', { contexts: ['TSFunctionType'] }],
     'jsdoc/require-returns-description': ['error', { contexts: ['TSFunctionType'] }],
+    'react/no-unstable-nested-components': ['error', { allowAsProps: true }],
```
Comment (cherniavskii) [inline .eslintrc.js:80]: To fix new eslint errors https://app.circleci.com/pipelines/github/mui/mui-x/26220/workflows/f133aad6-a5b3-405b-815d-86526f207584/jobs/150743

See https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/no-unstable-nested-components.md#rule-options for more details

**Comment by cherniavskii** [comment]
@m4theushw I think we can just remove the wrapper div. I'm not sure what was the original intention there

**Comment by alexfauquette** [comment]
It does not surprise me that setting a height too large implies breaking the doc layout. It's interesting to have access to this value quickly, to get the following interaction:

- Ok, it's fixed height demo
- What append if I modify the height?
- Ok, it removes the scroll bar and adds empty space between the last row and the footer
